perses/app/relative_point_mutation_setup.py

```diff
@@ -122,57 +106,73 @@ def __init__(self,
 
+        # Load the ligand, if present
         molecules = []
-        ligand_mol = createOEMolFromSDF(ligand_filename, index = ligand_index)
-        ligand_mol = generate_unique_atom_names(ligand_mol)
-        molecules.append(Molecule.from_openeye(ligand_mol,allow_undefined_stereo = False))
-        ligand_positions, ligand_topology = extractPositionsFromOEMol(ligand_mol),  forcefield_generators.generateTopologyFromOEMol(ligand_mol)
-        ligand_md_topology = md.Topology.from_openmm(ligand_topology)
-        ligand_n_atoms = ligand_md_topology.n_atoms
-
-        #now create a complex
-        complex_md_topology = receptor_md_topology.join(ligand_md_topology)
-        complex_topology = complex_md_topology.to_openmm()
-        complex_positions = unit.Quantity(np.zeros([receptor_n_atoms + ligand_n_atoms, 3]), unit=unit.nanometers)
-        complex_positions[:receptor_n_atoms, :] = receptor_positions
-        complex_positions[receptor_n_atoms:, :] = ligand_positions
-
-        #now for a system_generator
+        if ligand_filename:
+            if ligand_filename.endswith('.sdf'): # small molecule
+                ligand_mol = createOEMolFromSDF(ligand_filename, index=ligand_index)
+                ligand_mol = generate_unique_atom_names(ligand_mol)
+                molecules.append(Molecule.from_openeye(ligand_mol, allow_undefined_stereo=False))
+                ligand_positions, ligand_topology = extractPositionsFromOEMol(ligand_mol),  forcefield_generators.generateTopologyFromOEMol(ligand_mol)
+                ligand_md_topology = md.Topology.from_openmm(ligand_topology)
+                ligand_n_atoms = ligand_md_topology.n_atoms
+
+            elif ligand_filename.endswith('pdb'): # protein
+                ligand_pdbfile = open(ligand_filename, 'r')
+                ligand_pdb = app.PDBFile(ligand_pdbfile)
+                ligand_pdbfile.close()
+                ligand_positions, ligand_topology, ligand_md_topology = ligand_pdb.positions, ligand_pdb.topology, md.Topology.from_openmm(
+                    ligand_pdb.topology)
+                ligand_n_atoms = ligand_md_topology.n_atoms
+
+            # Now create a complex
+            complex_md_topology = protein_md_topology.join(ligand_md_topology)
+            complex_topology = complex_md_topology.to_openmm()
+            complex_positions = unit.Quantity(np.zeros([protein_n_atoms + ligand_n_atoms, 3]), unit=unit.nanometers)
+            complex_positions[:protein_n_atoms, :] = protein_positions
+            complex_positions[protein_n_atoms:, :] = ligand_positions
+
+        # Now for a system_generator
         self.system_generator = SystemGenerator(forcefields = forcefield_files,
                                                 barostat=barostat,
                                                 forcefield_kwargs=forcefield_kwargs,
+                                                periodic_forcefield_kwargs=periodic_forcefield_kwargs,
```

Comment:
this new? all tests are failing because the nonbonded method now has its own kwargs!